src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -0,0 +1,65 @@
+package xerus.monstercat.api
+
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+
+object Playlist {
+	var playlist: ObservableList<Song> = FXCollections.observableArrayList()
+	var currentTrack = 0
+	var repeat = false
+	var random: Boolean = false
+
+
+	fun next(): Song? {
+		when {
+			currentTrack + 1 < playlist.size -> currentTrack++
+			repeat -> currentTrack = 0
+			else -> return null
+		}
+		return playlist[currentTrack]
+	}
+
+	fun prev(): Song? {
+		if (currentTrack > 0) currentTrack--
+		return playlist[currentTrack]
+	}
+
+	fun select(index: Int): Song? {
+		if (index >= 0 && index < playlist.size) currentTrack = index
+		return playlist[currentTrack]
+	}
+
+	fun getTracks() = playlist
+	fun setTracks(playlist: MutableList<Song>) = this.playlist.addAll(playlist)
+
+	operator fun invoke() = getTracks()
```

Comment:
Please don't overuse `invoke`. It looks exactly like a constructor in Code and should be justified, otherwise the code becomes unreadable.

Reply:
Fixed